CoreScriptsRoot/LoadingScript.lua

```diff
@@ -762,3 +763,24 @@ game:GetService("ReplicatedFirst").RemoveDefaultLoadingGuiSignal:connect(handleR
 if game:GetService("ReplicatedFirst"):IsDefaultLoadingGuiRemoved() then
 	handleRemoveDefaultLoadingGui()
 end
+
+local UserInputServiceChangedConn;
+local function onUserInputServiceChanged(prop)
+	if prop == 'VREnabled' then
+		local UseVr = false
+		pcall(function() UseVr = UIS.VREnabled end)
+
+		if UseVr then
+			require(RobloxGui.Modules.LoadingScreen3D)
```

Comment:
There is some future-proofing at work here. There was some worry that the script would run before this property was set.